### PR TITLE
Allow potentially long names,tags and addresses to still be seen in UI

### LINKED ISSUE #106: [PE-D][Tester A] Long Tags are not Properly Displayed
When I tried to create a super-long tag for testing purposes, such as the one in the given diagram. 

The tag is cut off by the edge of the window and the person list is not horizontally scrollable for me to see the whole tag. Though the window can be resized in the horizontal direction, the tag maybe even longer than the user's screen.

I suppose this can be resolved in two ways, either by making a list horizontally scrollable or limiting the maximum length of the tag.

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: Yufannnn/ped#1

### LINKED ISSUE #120: [PE-D][Tester A] Long Names are Not Properly Dispalyed
Long Names of persons are cut off by the edges of the window for both the person list and the favorite person list so that I am not able to read the full name of the person.

I suppose this can be fixed by making a list horizontally scrollable or limiting the length of the name, or making an auto switch line function so that the name can be properly displayed

-------------
Labels: `severity.Low` `type.FunctionalityBug`
original: Yufannnn/ped#8

### LINKED ISSUE #129: [PE-D][Tester B] Name too long issue
If name is too long, you cant read the whole name. This also blocks the rank. 

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: homuntan02/ped#6

### PR DESCRIPTION
**Pull request by minosx31**
Enable text wrap for names,tags and address, with rank still remaining infront of name.

Before resize:
image

After resize:
image

Fixes #106

### CODE REVIEW

**Comment by rexcyrio** [comment]
What happens if I have 50 tags? Do all the 50 tags line up in a horizontal row such that all 50 tags will be displayed to the user? Will the tag be too narrow to read the tag name properly?

**Comment by minosx31** [comment]
> What happens if I have 50 tags? Do all the 50 tags line up in a horizontal row such that all 50 tags will be displayed to the user? Will the tag be too narrow to read the tag name properly?

The current implementation will try to squeeze all 50 tags in a single row but this would mean that the text inside is wrapped in such a way that it may not be readable. 

image



I could change it back such that the tags would still flow nicely and remain readable but would probably need to set a length restriction for the tags.
image

**Comment by rexcyrio** [comment]
For long names, we can simply let the text break to the next line -- is this possible for tags too (i.e., a very long tag name breaks to the next line)? Though, I am worried that it may not be viewed as a single tag item anymore cause of the line break *within* the light blue tag box.

Alternatively, maybe we can just set the max length for tags at 100 alphanumeric characters? Then we won't face the overflow issue anymore.

File: src/main/java/seedu/address/model/tag/Tag.java [hunk 1]
```diff
@@ -21,10 +22,20 @@ public class Tag {
...
-        checkArgument(isValidTagName(tagName), MESSAGE_CONSTRAINTS);
+        checkArgument(isValidTagLength(tagName), MESSAGE_LENGTH_CONSTRAINTS);
+        checkArgument(isValidTagName(tagName), MESSAGE_ALPHANUMERIC_CONSTRAINTS);
         this.tagName = tagName;
     }
 
+    /**
+     * Returns true if a given string is at most 100 characters long.
+     * @param test
+     * @return
```
Comment (rexcyrio) [inline src/main/java/seedu/address/model/tag/Tag.java:33]: Missing JavaDocs here

File: src/main/java/seedu/address/model/tag/Tag.java [hunk 2]
```diff
@@ -21,10 +22,20 @@ public class Tag {
...
     public Tag(String tagName) {
         requireNonNull(tagName);
-        checkArgument(isValidTagName(tagName), MESSAGE_CONSTRAINTS);
+        checkArgument(isValidTagLength(tagName), MESSAGE_LENGTH_CONSTRAINTS);
+        checkArgument(isValidTagName(tagName), MESSAGE_ALPHANUMERIC_CONSTRAINTS);
         this.tagName = tagName;
     }
 
+    /**
+     * Returns true if a given string is at most 100 characters long.
```
Comment (rexcyrio) [inline src/main/java/seedu/address/model/tag/Tag.java:31]: Is it `100 characters` or `60 characters`?

File: src/main/java/seedu/address/storage/JsonAdaptedTag.java [hunk 3]
```diff
@@ -39,8 +40,10 @@ public String getTagName() {
      * @throws IllegalValueException if there were any data constraints violated in the adapted tag.
      */
     public Tag toModelType() throws IllegalValueException {
-        if (!Tag.isValidTagName(tagName)) {
-            throw new IllegalValueException(Tag.MESSAGE_CONSTRAINTS);
+        if (!Tag.isValidTagLength(tagName)) {
+            throw new ParseException(Tag.MESSAGE_LENGTH_CONSTRAINTS);
```
Comment (rexcyrio) [inline src/main/java/seedu/address/storage/JsonAdaptedTag.java:44]: Do we want to throw a `ParseException` or an `IllegalValueException`?

In `JsonAdaptedPerson.java`:

```java
public Person toModelType() throws IllegalValueException {
    /* calls JsonAdaptedTag::toModelType() */
}
```

File: src/main/java/seedu/address/storage/JsonAdaptedTag.java [hunk 4]
```diff
@@ -39,8 +40,10 @@ public String getTagName() {
      * @throws IllegalValueException if there were any data constraints violated in the adapted tag.
      */
     public Tag toModelType() throws IllegalValueException {
```
Comment (rexcyrio) [inline src/main/java/seedu/address/storage/JsonAdaptedTag.java:42]: See my comment below -- both the JavaDocs and method signature `throws XYZException` may need to be updated.

**Review by rexcyrio — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/address/model/tag/Tag.java [hunk 5]
```diff
@@ -21,10 +22,20 @@ public class Tag {
...
+     * @param test
+     * @return
+     */
+    public static boolean isValidTagLength(String test) {
+        return test.length() <= 60;
+    }
```
[the remark this answers is not in the document]
Comment (minosx31) [reply src/main/java/seedu/address/model/tag/Tag.java:37]: Yeah good suggestion! Will add them in the next commit 😃

**Review by rexcyrio — CHANGES_REQUESTED** [review]

**Review by rexcyrio — APPROVED** [review]
LGTM :+1:

**Pull request merged**